etc/profile.d/base.sh

```diff
@@ -16,23 +16,29 @@ metal() {
 }
 
 if [ "$ZSH_VERSION" ]; then
-  export metal
+    export metal
 else
-  export -f metal
+    export -f metal
 fi
 alias met=metal
 
-# XXX Disabled as completion not done yet for new Metalware.
-# if [ "$BASH_VERSION" ]; then
-#     _metal() {
-#         local cur="$2" prev="$3" cmds opts
+if [ "$BASH_VERSION" ]; then
+    _metal() {
+        local cur="$2" prev="$3" cmds opts
 
-#         COMPREPLY=()
+        path=$( IFS=$'/'; echo "${COMP_WORDS[*]}" | sed "s/^metal\/\|$cur$//g")
+        cur_dir="/opt/metalware/src/commands/$path"
 
-#         cmds=$(ls /opt/metalware/lib/actions)
+        if [ -d "$cur_dir" ]; then
+            cmds=$(ls $cur_dir | sed s/\.rb//g)
+        fi
 
-#         COMPREPLY=( $(compgen -W "$cmds" -- "$cur") )
-#     }
+        # Additional bash commands, ideally these will be migrated to ruby
+        if [ "$path" == "/" ]; then
+            cmds="$cmds power console"
```

Comment:
These commands aren't being autocompleted for me after a fresh install - do they complete for you?